src/lib/eslint/BaseEslintEngine.ts

```diff
@@ -230,19 +236,21 @@ export abstract class BaseEslintEngine implements RuleEngine {
 	}
 
 	/* eslint-disable @typescript-eslint/no-explicit-any */
-	private selectRelevantRules(rules: Rule[]): Record<string,any> {
+	private async selectRelevantRules(rules: Rule[]): Promise<Record<string,any>> {
 		const filteredRules = {};
-		let ruleCount = 0;
-
-		for (const rule of rules) {
-			// Find if a rule is relevant
-			if (rule.engine === this.getName()) {
```

Comment:
The PMD engine doesn't check this, and the DefaultRuleManager filters out the rules already. The caller should be responsible for this.